scripts/host-check

```diff
@@ -67,6 +67,13 @@ from typing import (
 # Globals
 # -----------------------------------------------------------------------------
 
+
+# Exit codes
+EXIT_SUCCESS = 0
+EXIT_ERROR = 10
+EXIT_WARNING = 11
```

Comment:
I'm not a huge fan of these specific error codes - in fact it's making me question whether we should use different exit codes at all!

What are trying to achieve here?  Currently we have the following check states:
* Success
* Fail
* Error (tried to perform the check but could not)
* Skip (did not perform the check because a dependent check was fail/error).
* Warn (i.e., YMMV)

The result of each check is reported inline, but these are easy to miss/glance over, and so we have an in-your-face summary at the bottom.

Currently we distinguish between three summary cases:
* Success
* Fail/Warn (in this case we say that the platform is NOT supported)
* Error

And the idea of this PR is to distinguish between Fail and Warn; e.g., in the Warn case then the user is free to give it a try.

I think indicating to the user the "summary result" is reasonable.  I'm less sure of the value of having separate exit codes:
* Exit code conventions are not well-established (beyond 0, 1, and 2)
* [Therefore] this is a documentation burden
* We should encourage the same action in each nonzero case: fix the Fail/Warn/Error until we can categorically say the platform is supported.
* This is a potentially breaking change ... e.g., if users are explicitly checking for exit code 1 as failure (rather than just nonzero).

Might be worth canvassing @paulo-cisco's opinion..